source/libs/api.ts

```diff
@@ -67,15 +75,17 @@ function fetch4(query, personalToken) {
 }
 
 // Main function: handles cache, options, errors
-async function call(fetch, query, options = {accept404: false}) {
+async function call(fetch: FetchStrategy, query: string, options: FetchOptions = {accept404: false}) {
 	if (cache.has(query)) {
 		return cache.get(query);
 	}
 
-	const {personalToken} = await new OptionsSync().getAll();
+	const {personalToken} = await new OptionsSync().getAll<{personalToken: string}>();
```

Comment:
This can probably also use the `DefineOptions` I mentioned.

Reply:
I don't think it'd work with `DefineOptions` (now called `Definitions`). The shape doesn't match

```
export interface Definitions {
	defaults: Options;
	/**
	 * A list of functions to call when the extension is updated.
	 */
	migrations: Migration[];
}
```

Maybe you meant `Options`?

`type Options = Record<string, string | number | boolean>;`


I can set it to it, but need to export it in the types (just updated my PR with this as @sindresorhus asked for it to be exported). If you type it to `Options`, then it needs to be cast to string when accessing it later, which is totally fine.

![image](https://user-images.githubusercontent.com/833231/52320196-f6df2680-299b-11e9-8415-e4c534668dbb.png)

In fact, it type type probably shouldn't be a generic as it's always returning `Promise<Options>` come to think of it. I'll update the types.

Reply:
See https://github.com/bfred-it/webext-options-sync/pull/15/commits/6c916f1a7a853f019521ef048ccaa14fa3ea04c9

Reply:
Yeah that's fine. I wonder if those types need to be namespaced somehow, `Options`, `Definitions` and `Migrations` make sense in the module but how does it work globally? In your screenshot, TypeScript knows that `Options` comes from `webext-options-sync` and not from somewhere else?